core-api/src/worker.rs

```diff
@@ -357,6 +362,12 @@ pub trait SlotSupplier {
     fn available_slots(&self) -> Option<usize> {
         None
     }
+
+    /// Returns a human-friendly identifier describing this supplier implementation for
+    /// diagnostics and telemetry.
+    fn slot_supplier_kind(&self) -> String {
+        type_name::<Self>().to_string()
```

Comment:
This is just going to be some gobbledygook like `lang_bridge_name::CustomSlotSupplierImpl` or something. I would just default it to `"Custom"`